src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/Extensions/SyntaxGeneratorExtensions_Negate.cs

```diff
@@ -229,7 +241,10 @@ private static SyntaxNode GetNegationOfIsPatternExpression(SyntaxNode isExpressi
             if (syntaxFacts.SupportsNotPattern(semanticModel.SyntaxTree.Options))
             {
                 // We do support 'not' patterns.  So attempt to push a 'not' pattern into the current is-pattern RHS.
-                negatedPattern = generator.Negate(generatorInternal, pattern, semanticModel, cancellationToken);
+                // If the value isn't a Boolean and the pattern `is true/false`, swapping to `is false/true` is incorrect since non-Booleans match neither.
```

Comment:
nit: it's often good to add an example to demonstrate the problem.  but not required.

Reply:
We can just add one if you'd like.

```suggestion
                // If the value isn't a Boolean and the pattern `is true/false`, swapping to `is false/true` is incorrect since non-Booleans match neither.
                // As an example, `!(new object() is true)` is equivalent to `new object() is not true` but not `new object() is false`.
```

Reply:
perfect, thanks!